go/libraries/doltcore/sqle/table_editor_fk_test.go

```diff
@@ -359,7 +371,266 @@ ALTER TABLE three ADD FOREIGN KEY (v1, v2) REFERENCES two(v1, v2) ON DELETE CASC
 	}
 }
 
-func assertTableEditorRows(t *testing.T, fk_dEnv *env.DoltEnv, root *doltdb.RootValue, expected []sql.Row, tableName string) {
+func TestTableEditorSelfReferentialForeignKeyRestrict(t *testing.T) {
+	ctx := context.Background()
+	root := fk_initialRoot
+
+	sequentialTests := []struct {
+		statement   string
+		currentTbl  []sql.Row
+		expectedErr bool
+	}{
+		{
+			"ALTER TABLE parent ADD CONSTRAINT fk_named FOREIGN KEY (v2) REFERENCES parent(v1);",
+			[]sql.Row{},
+			false,
+		},
+		{
+			"INSERT INTO parent VALUES (1,1,1), (2, 2, 1), (3, 3, NULL);",
+			[]sql.Row{{1, 1, 1}, {2, 2, 1}, {3, 3, nil}},
+			false,
+		},
+		{
+			"UPDATE parent SET v1 = 1 WHERE id = 1;",
+			[]sql.Row{{1, 1, 1}, {2, 2, 1}, {3, 3, nil}},
+			false,
+		},
+		{
+			"UPDATE parent SET v1 = 4 WHERE id = 3;",
+			[]sql.Row{{1, 1, 1}, {2, 2, 1}, {3, 4, nil}},
+			false,
+		},
+		{
+			"DELETE FROM parent WHERE id = 3;",
+			[]sql.Row{{1, 1, 1}, {2, 2, 1}},
+			false,
+		},
+		{
+			"DELETE FROM parent WHERE v1 = 1;",
+			[]sql.Row{},
+			true,
+		},
+		{
+			"UPDATE parent SET v1 = 2;",
```

Comment:
What about a test where you are setting the child and parent column both to the same value in the same statement?

Reply:
In an `UPDATE`? I can add that.